### PR TITLE
Added github workflows to build Ubuntu/Debian and Mac OSX packages.

### PR DESCRIPTION
**Pull request by t0mpr1c3**
Referencing https://github.com/AllYarnsAreBeautiful/ayab-desktop/issues/375, https://github.com/AllYarnsAreBeautiful/ayab-desktop/issues/310

The Fedora and Arch workflows require GPG keys to build the Docker VMs, ~~which I guess need to be written to the `secrets` settings in the repo. I don't have the permissions to do that.~~ I found a way to write the GPG password and build the VMs on the github runner using `fbs buildvm`, but running them with `fbs runvm` failed. I succeeded in building Arch and Fedora packages locally, though.

### CODE REVIEW

**Comment by chris007de** [comment]
Thanks a lot for your efforts! I'll need some time to work through this and will give you feedback.
But what I can already tell from skipping through the changes: We'll probably not wanna store the private key for signing the binaries publicly available in our repository :)
I'll check how to deposit such a key as a github secret.

File: .github/workflows/build-fedora.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      run: echo "::set-env name=GIT_TAG::$(git describe --tags)"
+    - name: Set PACKAGE_VERSION
+      run: |
+        echo ${{env.GIT_TAG}} > src/main/resources/base/ayab/package_version
+        sed -i 's/PACKAGE_VERSION/${{env.GIT_TAG}}/' src/build/settings/base.json
+        cat src/main/resources/base/ayab/package_version
+    - name: Set up Python 3.6
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.6
```
Comment (chris007de) [inline .github/workflows/build-fedora.yml:31]: Perhaps make also use of `strategy: matrix: python-version...`? -> ${{ matrix.python-version }}

File: .github/workflows/build-ubuntu.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        echo "::set-env name=GIT_TAG::$(git describe --tags)"
+    - name: Set PACKAGE_VERSION
+      run: |
+        echo ${{env.GIT_TAG}} > src/main/resources/base/ayab/package_version
+        sed -i 's/PACKAGE_VERSION/${{env.GIT_TAG}}/' src/build/settings/base.json
+        cat src/main/resources/base/ayab/package_version
+    - name: Set up Python 3.6
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.6
```
Comment (chris007de) [inline .github/workflows/build-ubuntu.yml:32]: see above, define python-version in matrix and make use of ${{ matrix.python-version }}

File: .github/workflows/build-arch.yml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+on:
+  push:
+    branches:
+      - master
+      # - '*-dev'
+
+jobs:
+  build:
+    runs-on: ubuntu-16.04
```
Comment (chris007de) [inline .github/workflows/build-arch.yml:14]: Should be running with ubuntu-latest as well, shouldn't it?

File: .github/workflows/build-arch.yml [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        python-version: ${{ matrix.python-version }}
+    - name: Use old pip version
+      run: python3 -m pip install pip==18.1
+    - name: Update setuptools
+      run: python3 -m pip install --upgrade setuptools
+    - name: Install fbs
+      run: python3 -m pip install fbs==0.8.6
+    - name: Create Docker image
+      run: |
+        echo '{"gpg_pass": "password"}' > src/build/settings/secret.json
```
Comment (chris007de) [inline .github/workflows/build-arch.yml:42]: I can store the gpg password as secred in the github repo (see https://docs.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets#creating-encrypted-secrets-for-a-repository), please send it to me over a secure channel.

File: .github/workflows/build-fedora.yml (hunk 1 above)
Comment (t0mpr1c3) [reply .github/workflows/build-fedora.yml:31]: OK

File: .github/workflows/build-ubuntu.yml (hunk 2 above)
Comment (t0mpr1c3) [reply .github/workflows/build-ubuntu.yml:32]: OK

File: .github/workflows/build-arch.yml (hunk 3 above)
Comment (t0mpr1c3) [reply .github/workflows/build-arch.yml:14]: For the time being I am keeping the Linux scripts on Ubuntu 16.04 for the sake of compatibility as this is still a supported version. Anything that works on 16.04 should also work on 18.04, but the reverse is not necessarily true.

File: .github/workflows/build-arch.yml (hunk 4 above)
Comment (t0mpr1c3) [reply .github/workflows/build-arch.yml:42]: This is a dummy password anyway! We can do this when the scripts are more mature.

**Comment by t0mpr1c3** [comment]
> * In production, I'd propose to run the workflows only on when a tag is pushed (see https://github.com/trappitsch/fbs-release-github-actions/blob/master/workflows/release-ubuntu.yml#L5). Generating installers (and Github releases) is not necessary for each and every push in my opinion.

Yes.

>   For now, during development, you could define to run only on push to branch `1.0.0-build` so we see how everything behaves.

OK.

> * Please add the steps for creating releases and upload the results (just as shown in the https://github.com/trappitsch/fbs-release-github-actions examples)

Will do.

> * If we switch our changelog format to https://keepachangelog.com/, we could even make use of https://github.com/marketplace/actions/changelog-reader later ;)
> * We can remove ayab_logo_rc.py now, as it is generated automatically, right?

Yes.

**Comment by t0mpr1c3** [comment]
In the latest commit have removed `build-arch.yml` and `build-fedora.yml` as works-in-progress. I now have a script to build an AppImage that compiles everything except the audio packages (which are not packaged as wheels -- the `setup.py` scripts break). So we can have a silent cross-Linux version.

File: linux-build/appimage/ayab.desktop [hunk 5]
```diff
@@ -0,0 +1,10 @@
+[Desktop Entry]
+Type=Application
+Name=ayab
+Exec=ayab
+Comment=All Yarns Are Beautiful
+Icon=ayab
+Categories=Graphics;
+Terminal=false
+X-AppImage-Version=v0.95-104-gc97f0b6
```
Comment (chris007de) [inline linux-build/appimage/ayab.desktop:9]: I guess we should update this field dynamically, as we do with `package_version`...time for another script? :D

File: requirements.txt [hunk 6]
```diff
@@ -1,14 +1,15 @@
-pip==18.1
+#pip==18.1
```
Comment (chris007de) [inline requirements.txt:1]: Is pip==18.1 no more hard requirement?

File: linux-build/appimage/ayab.desktop (hunk 5 above)
Comment (t0mpr1c3) [reply linux-build/appimage/ayab.desktop:9]: I think this field is updated automatically.

File: requirements.txt (hunk 6 above)
Comment (t0mpr1c3) [reply requirements.txt:1]: Correct.

**Comment by t0mpr1c3** [comment]
I put the Ubuntu, Mac, and Windows release scripts in a single file. It's definitely better this way, because they release assets can all be attached to the same release tag. I haven't tested the Mac `.dmg`, but the other two seem to work. 

I think we can close this PR. If I can get the AppImage running I'll raise a separate PR.

**Pull request merged**